driver/src/contracts/stablex_contract.rs

```diff
@@ -0,0 +1,162 @@
+#[cfg(test)]
+extern crate mock_it;
+
+use dfusion_core::models::{AccountState, Order};
+
+use web3::contract::Options;
+use web3::futures::Future;
+use web3::types::{H160, U128, U256};
+
+use crate::error::DriverError;
+
+use super::base_contract::BaseContract;
+
+use std::env;
+use std::fs;
+
+type Result<T> = std::result::Result<T, DriverError>;
+
+#[allow(dead_code)] // TODO - remove once used
+struct StableXContractImpl {
+    base: BaseContract
+}
+
+#[allow(dead_code)] // TODO - remove once used
+impl StableXContractImpl {
+    pub fn new() -> Result<Self> {
+        let contract_json = fs::read_to_string("dex-contracts/build/contracts/StablecoinConverter.json");
+        let address = env::var("STABLEX_CONTRACT_ADDRESS");
+        Ok(
+            StableXContractImpl {
+                base: BaseContract::new(address?, contract_json?)?
+            }
+        )
+    }
+}
+
+pub trait StableXContract {
+    fn get_current_auction_index(&self) -> Result<U256>;
+    fn get_auction_data(&self, _index: U256) -> Result<(AccountState, Vec<Order>)>;
+
+    fn submit_solution(
+        &self,
+        batch_index: U256,
+        owners: Vec<H160>,
+        order_ids: Vec<U128>,
+        volumes: Vec<U128>,
+        prices: Vec<U128>,
+        token_ids_for_price: Vec<U128>,
+    ) -> Result<()>;
+}
+
+impl StableXContract for StableXContractImpl {
+    fn get_current_auction_index(&self) -> Result<U256> {
+        self.base.contract
+            .query("getCurrentStateIndex", (), None, Options::default(), None)
+            .wait()
+            .map_err(DriverError::from)
+    }
+
+    fn get_auction_data(&self, _index: U256) -> Result<(AccountState, Vec<Order>)> {
+        unimplemented!();
+    }
+
+    fn submit_solution(
+        &self,
```

Comment:
I wonder if it would be better to use `Solution` and Vec<Order> as an exposing abstraction level here and have the conversion logic into all these separate arrays happen inside the contract wrapper.

What do you think?

Reply:
I think I understand your reference to `Solution` as in passing a `Solution` type into this function, but I don't know what you mean by `Vec` here... unless there is additional information I am forgetting about. I would generally agree with this suggestion, but probably would want to make these changes in the following PR (since it is currently unimplemented and would not affect this meaningfully).

Reply:
So yes, @fleupold, overall I can completely agree with these suggestions but would like to carry over this discussion to the next PR where submit solution will be implemented if that's alright with you!

Reply:
Ok, I now see that this PR would make the most sense for these suggested changes, however, I am not entirely sure I fully understand the proposal. I suppose this would look something like 

```rs
    fn submit_solution(
        &self,
        batch_index: U256,
        owners: Vec<H160>,
        order_ids: Vec<U128>,
        solution: Solution,
        token_ids_for_price: Vec<U128>,
    ) -> Result<()>;
```
but in this case I have only reduced prices and volumes. I will have to look closer to see what else can be logically packed into a `Solution` type.

Reply:
I'd suggest for the method to look like:

```
fn submit_solution(
        &self,
        batch_index: U256,
        orders: Vec<Order>,
        solution: Solution,
    ) -> Result<()>;
```

Owners should be extractable from orders and token_ids should be extractable from the solution.

Reply:
Ya cool! Thanks.. I had already determined we wouldnb't need `token_ids` but wasn't yet all the way there with passing orders and getting owners from them. Thanks!